test/WebJobs.Script.Tests.Integration/ApplicationInsights/ApplicationInsightsTestFixture.cs

```diff
@@ -1,40 +1,150 @@
-            // we're hitting a local HttpListener, it can be anything.
-            ScriptSettingsManager.Instance.ApplicationInsightsInstrumentationKey = TestChannelLoggerFactoryBuilder.ApplicationInsightsKey;
-        }
+        private readonly ScriptSettingsManager _settingsManager;
+        private readonly HttpConfiguration _config = new HttpConfiguration();
 
         public ApplicationInsightsTestFixture(string scriptRoot, string testId)
-            : base(scriptRoot, testId)
         {
+            _settingsManager = ScriptSettingsManager.Instance;
+
+            HostSettings = new WebHostSettings
+            {
+                IsSelfHost = true,
+                ScriptPath = Path.Combine(Environment.CurrentDirectory, scriptRoot),
+                LogPath = Path.Combine(Path.GetTempPath(), @"Functions"),
+                SecretsPath = Environment.CurrentDirectory, // not used
+                LoggerFactoryBuilder = new TestLoggerFactoryBuilder(Channel),
+                IsAuthDisabled = true
+            };
+            WebApiConfig.Register(_config, _settingsManager, HostSettings);
+
+            var resolver = _config.DependencyResolver;
+            var hostConfig = resolver.GetService<WebHostResolver>().GetScriptHostConfiguration(HostSettings);
+
+            _settingsManager.ApplicationInsightsInstrumentationKey = TestChannelLoggerFactoryBuilder.ApplicationInsightsKey;
+
+            InitializeConfig(hostConfig);
+
+            var httpServer = new HttpServer(_config);
+            HttpClient = new HttpClient(httpServer)
+            {
+                BaseAddress = new Uri("https://localhost/")
+            };
+
+            WaitForHost();
         }
 
         public TestTelemetryChannel Channel { get; private set; } = new TestTelemetryChannel();
 
-        protected override void InitializeConfig(ScriptHostConfiguration config)
-        {
-            var builder = new TestChannelLoggerFactoryBuilder(Channel);
-            config.HostConfig.AddService<ILoggerFactoryBuilder>(builder);
+        public WebHostSettings HostSettings { get; private set; }
 
-            // turn this off as it makes validation tough
-            config.HostConfig.Aggregator.IsEnabled = false;
+        public HttpClient HttpClient { get; private set; }
 
+        protected void InitializeConfig(ScriptHostConfiguration config)
+        {
             config.OnConfigurationApplied = c =>
             {
-                // Overwrite the generated function whitelist to only include one function.
-                c.Functions = new[] { "Scenarios" };
+                // turn this off as it makes validation tough
+                config.HostConfig.Aggregator.IsEnabled = false;
+
+                // Overwrite the generated function whitelist to only include two functions.
+                c.Functions = new[] { "Scenarios", "HttpTrigger-Scenarios" };
             };
         }
+
+        private void WaitForHost()
```

Comment:
nit: I feel like I've seen these in other fixtures - can we move them to a utility?

Reply:
Done.